lua/wire/client/text_editor/pickcolor_browser.lua

```diff
@@ -0,0 +1,127 @@
+--[[
+
+	ColorPicker for Expression2
+	Allows faster color picking whithin the editor.
+
+	Supported Color Format for Editors:
+	GPU: RGB like: "color White, 255,255,255"
+	E2: RGB like: "vec(255)"
+
+]]--
+
+-- Language Features
+language.Add("pickcolorlang.copyvec", "Use vec(r, g, b) structure")
+language.Add("pickcolorlang.copyalpha", "Use vec4(r, g, b, a) structure.")
+
+-- Convars
+CreateClientConVar("wire_expression2_pickcolor_enable", 1, true, false)
+CopyWithVecString = CreateClientConVar("wire_expression2_colorwithvec", 1, true, false)
+CopyWithAlpha = CreateClientConVar("wire_expression2_copyalpha", 0, true, false)
+
+-- Follows sound_browser pattern
+-- Invalidates the panel on start
+local PickColorPanel = nil
+
+local function GetCurrentEditorType()
+	return GetEditorType
+end
+
+-- Checks the Color and Checkboxes and copies to clipboard.
+local function ColorToClipboard(colorStr)
+
+	local clrCopy
+	local etype = GetCurrentEditorType()
+	local cfmt = ", " -- comma format
+
+	if etype == "E2" then
+		if CopyWithVecString:GetBool() and CopyWithAlpha:GetBool() then
+			clrCopy = "vec4("..colorStr.r..cfmt..colorStr.g..cfmt..colorStr.b..cfmt..colorStr.a..")"
+
+		elseif CopyWithVecString:GetBool() and not CopyWithAlpha:GetBool() then
+			clrCopy = "vec("..colorStr.r..cfmt..colorStr.g..cfmt..colorStr.b..")"
+
+		elseif not CopyWithVecString:GetBool() and CopyWithAlpha:GetBool() then
+			clrCopy = colorStr.r..cfmt..colorStr.g..", "..colorStr.b..cfmt..colorStr.a
+
+		elseif not CopyWithVecString:GetBool() and not CopyWithAlpha:GetBool() then
+			clrCopy = colorStr.r..cfmt..colorStr.g..cfmt..colorStr.b
+		end
+
+	elseif etype == "GPU" then
+		cfmt = ","
```

Comment:
Does the GPU support spaces between the commas? I'd be very surprised if it didn't.